File: hgdb/java/read/hgdb_read.java
// -------------------------------------------------------------- 
//
//	hgdb_read.java
//
//						Feb/01/2012
//
// --------------------------------------------------------------
import	java.util.HashMap;
// --------------------------------------------------------------
public class hgdb_read
{

// --------------------------------------------------------------
public static void main(String[] args)
{
	System.out.println ("*** 開始 ***");

	String databaseLocation = args[0];

	HashMap <String, HashMap<String,String>> dict_aa
		= hgdb_manipulate.hgdb_to_dict_proc (databaseLocation);

	text_manipulate.dict_display_proc (dict_aa);

	System.out.println ("*** 終了 ***");
}

// --------------------------------------------------------------
}
// --------------------------------------------------------------

File: hgdb/scala/read/hgdb_read.scala
// -------------------------------------------------------------- 
//
//	hgdb_read.scala
//
//						Oct/07/2011
//
// --------------------------------------------------------------
import org.hypergraphdb._
import org.hypergraphdb.query._

import org.hypergraphdb.HGHandle
import org.hypergraphdb.HGSearchResult
import org.hypergraphdb.HyperGraph
import org.hypergraphdb.query.AtomValueCondition
import org.hypergraphdb.query.ComparisonOperator
import org.hypergraphdb.query.HGQueryCondition
import org.hypergraphdb.query.AtomTypeCondition


import	java.util.HashMap
// --------------------------------------------------------------
object hgdb_read
{

// --------------------------------------------------------------
def main (args: scala.Array[String])
{
	println ("*** 開始 ***")

	val databaseLocation = "/var/tmp/hgdb/cities_db"

	val graph = new HyperGraph(databaseLocation)

	val condition = new AtomTypeCondition ()
//	val condition = new AtomTypeCondition (HashMap.class)
//	val condition = new AtomTypeCondition (java.util.HashMap.getClass ())
//	val condition = new AtomTypeCondition (java.util.HashMap)
//	val condition = new AtomTypeCondition (java.util.HashMap.class)
//	val condition = new AtomTypeCondition (Class[HashMap])

	val rs = graph.find(condition)

	while (rs.hasNext())
		{
		val current = rs.next()
		val result:HashMap[String,Object] = graph.get(current)

		System.out.print (result.get ("id") + "\t")
//		System.out.print (result.get ("id") + "\t")
		System.out.print (result.get ("name") + "\t")
		val ppx = result.get ("population")
		System.out.print (ppx + "\t")

		System.out.println (result.get ("date_mod"))
                }

	rs.close()

	println ("*** 終了 ***")
}
// --------------------------------------------------------------
}
// --------------------------------------------------------------
